src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+
+#pragma warning disable 0414
+
+namespace System.Reflection.Tests
+{
+    public class PropertyInfoGetValueTests
+    {
+        public static void TestGetSet_PropertyA()
+        {
+            string propertyName = "PropertyA";
+            MyCoVariantTest obj = new MyCoVariantTest();
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj);
+
+            Assert.NotNull(value);
+
+            string[] strs = new string[1];
+            strs[0] = "hello";
+
+            //set value
+            pi.SetValue(obj, strs);
+            value = pi.GetValue(obj);
+
+            string[] strs2 = (string[])value;
+
+            Assert.Equal(strs2[0], strs[0]);
+        }
+
+
+        [Fact]
+        public static void TestGetSet_PropertyB()
+        {
+            string propertyName = "PropertyB";
+            MyCoVariantTest obj = new MyCoVariantTest();
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj);
+
+            Assert.Null(value);
+
+            string[] strs = new string[1];
+            strs[0] = "hello";
+
+            //set value
+            pi.SetValue(obj, strs);
+            value = pi.GetValue(obj);
+
+            string[] strs2 = (string[])value;
+
+            Assert.Equal(strs2[0], strs[0]);
+        }
+
+        [Fact]
+        public static void TestGetSet_Name()
+        {
+            string propertyName = "Name";
+            InterfacePropertyImpl obj = new InterfacePropertyImpl();
+            PropertyInfo pi = GetProperty(typeof(InterfacePropertyImpl), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj, (Object[])null);
+
+            Assert.Null(value);
+
+            //set value
+            string strs1 = "hello";
+
+            pi.SetValue(obj, strs1);
+            value = pi.GetValue(obj);
+            string strs2 = (string)value;
+            Assert.Equal(strs2, strs1);
+        }
+
+        [Fact]
+        public static void TestGet_PropertyC()
```

Comment:
You could try the following for the GetValue tests to eliminate boilerplate and make the tests easier to follow
```
public static IEnumerable<object[]> GetValue_TestData()
{
    yield return new object[] { typeof(MyCoVariantTest), "PropertyC", new MyCoVariantTest(), new object[] { 1, "2" }, null };
    // ...
}

[Theory]
[MemberData(nameof(GetValue_TestData)]
public void GetValue(Type parentType, string propertyName, object instance, object[] index, object expectedValue)
{
	PropertyInfo propertyInfo = GetProperty(parentType, propertyName);
	if (index == null)
	{
		Assert.Equal(expectedValue, propertyInfo.GetValue(instance));
	}
	Assert.Equal(expectedValue, propertyInfo.GetValue(instance, index));
}
```

I'm sure SetValue can have similar template